src/ports/postgres/modules/deep_learning/test/input_data_preprocessor.sql_in

```diff
@@ -51,15 +51,12 @@ SELECT training_preprocessor_dl(
 SELECT assert(count(*)=4, 'Incorrect number of buffers in data_preprocessor_input_batch.')
 FROM data_preprocessor_input_batch;
 
-SELECT assert(array_upper(independent_var, 2)=6, 'Incorrect buffer size.')
+SELECT assert(independent_var_shape ='[1:4][1:6]', 'Incorrect buffer size.')
 FROM data_preprocessor_input_batch WHERE buffer_id=0;
 
-SELECT assert(array_upper(independent_var, 1)=5, 'Incorrect buffer size.')
+SELECT assert(independent_var_shape ='[1:5][1:6]', 'Incorrect buffer size.')
```

Comment:
In test cases like these, it would be good to validate that the length of `independent_var` (in addition to checking that the `independent_var_shape` field is accurate).  I think we could easily have a regression where the first ends up wrong but the second is right.

For example, for these two cases we could do this by asserting `length(independent_var = 24` and `length(independent_var) = 30`.

Reply:
(possibly, `octet_length` may be safer... but I think `length` should be okay too)